### ISSUE TITLE
Request completionItem/resolve failed error

### ISSUE THREAD

**Issue by johno** [mdx-js/mdx-analyzer#301]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

Latest (1.2.2)

### Steps to reproduce

It seems to happen intermittently, and haven't fully figured out why/where it occurs. Here's an example of an MDX document that reproduces the error:

```
# Example document

- [Link](/a)
- [Link](/a)
- [Link](/a)

## Below 

Typing here causes the error CURSOR LOCATION

- [Link](/a)
```

### Expected behavior

Doesn't error, or at the very least does so silently.

### Actual behavior

It errors, and once it gets going, does so repeatedly.

### Screenshot

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/1424573/219979888-1ef3c1ed-e194-4617-b40e-95285ca8d1b9.png">

### Logs

```sh
[Error - 3:09:56 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:11:48 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:12:16 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:12:28 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:12:32 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:14:39 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:20:22 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
[Error - 3:44:56 PM] Request completionItem/resolve failed.
  Message: Request completionItem/resolve failed with message: Cannot read properties of undefined (reading 'kind')
  Code: -32603 
```

### Runtime

Node v18

### Package manager

yarn v1

### OS

macOS

### Build and bundle tools

webpack, Next.js

---

Thank you for your work on this project, it's great! 🙏 🙏 💟

**Comment by remcohaszing** [comment]
I tried it. It’s doesn’t happen after every keystroke, but I am able to reproduce this bug.

Thanks for reporting!

**Comment by remcohaszing** [comment]
Released in `vscode-mdx@1.5.0`, `@mdx-js/language-server@0.2.0` :tada:
